### PR TITLE
feat: Make various data publicly accessible

### PR DESCRIPTION
**Pull request by doug-q**

### CODE REVIEW

File: quantinuum-hugr/src/types/type_param.rs [hunk 1]
```diff
@@ -246,6 +251,12 @@ impl TypeArg {
     }
 }
 
+impl TypeArgVariable {
```
Comment (doug-q) [inline quantinuum-hugr/src/types/type_param.rs:254]: There's no docstring here, why isn't CI catching this?

Comment (aborgna-q) [reply quantinuum-hugr/src/types/type_param.rs:254]: Uh, seems I missed setting `lints.workspace = true` in the package's Cargo.toml when I created the workspace...

I can fix it in a separate PR, or you can add
```toml
[lints]
workspace = true
```
in `quantinuum-hugr/Cargo.toml`.

**Review by aborgna-q — APPROVED** [review]
:+1:

Just add the missing doc.

File: quantinuum-hugr/src/types/type_param.rs (hunk 1 above)
Comment (aborgna-q) [reply quantinuum-hugr/src/types/type_param.rs:254]: See #876

**Pull request merged**